opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -814,6 +811,31 @@ def _contains_something_valuable(thing):
     return True
 
 
+def attach_markers(timeline):
+    '''
+    Search for markers and attach them to their corresponding track
+    '''
+    markers = []
+
+    # Get all markers in the given unsimplified timeline. Markers come in as non
+    # audio or video tracks
+    for child in list(timeline.each_child()):
+        if isinstance(child, otio.schema.Track) and child.markers:
+            markers.extend(child.markers)
+            child.parent().remove(child)
+
+    # Add marker(s) to corresponding video or audio track
+    for child in timeline.each_child():
+        if (isinstance(child, otio.schema.Track) and child.kind in
+                [otio.schema.TrackKind.Audio, otio.schema.TrackKind.Video]):
+            slot_id = child.metadata.get('AAF').get('SlotID')
+            for m in markers:
+                if str(slot_id) in m.metadata.get('AAF').get('DescribedSlots'):
```

Comment:
Could you build a dictionary mapping the marker described slots to a list of markers?  That way you could iterate through the tracks and just add all the relevant markers?